commands/auth_utils_test.go

```diff
@@ -85,6 +86,29 @@ func TestResolveConfigFilePath(t *testing.T) {
 		p := resolveConfigFilePath()
 		require.Equal(t, expected, p)
 	})
+
+	t.Run("should create the user-defined config file path if one is set", func(t *testing.T) {
+		tmp, _ := ioutil.TempDir("", "mmctl-")
+		defer os.RemoveAll(tmp)
+
+		testUser.HomeDir = "path/should/be/ignored"
+		SetUser(testUser)
+		extraDir := "extra"
+
+		expected := filepath.Join(tmp, extraDir, "config.json")
+
+		err := os.Setenv("XDG_CONFIG_HOME", "path/should/be/ignored")
+		require.NoError(t, err)
+		viper.Set("config", expected)
```

Comment:
Should `--config` be a path to file or can be a directory as well? If yes, can we add a test where we just supply a custom dir ending with `_config` and ensure that a redundant directory does not get created?

Reply:
It is a file path, the old flag (`--config-path`) was expecting it to be a directory and filename was being determined by the tool itself. That was a little bit problematic approach therefore we replaced it (with `--config`, in fact we emphasized that flag expects a file in the docs and help text).

Reply:
Great :)